File: my-folder/0131-palindrome-partitioning/solution.cpp
class Solution {
public:
    bool checkpal(string s,int i,int j){
        while(i<=j){
            if(s[i++]!=s[j--]) return false;
        }
        return true;
    }
    void recursion(string s, vector<vector<string>> &ans,vector<string> temp,int i){
        if(i==s.size()){
            ans.push_back(temp);
            return;
        }
        for(int ind = i;ind<s.size();ind++){
            if(checkpal(s,i,ind)){
                temp.push_back(s.substr(i,ind-i+1));
                recursion(s,ans,temp,ind+1);
                temp.pop_back();

            }
        }
    }
    vector<vector<string>> partition(string s) {
        vector<vector<string>> ans;
        recursion(s,ans,{},0);
        return ans;
        
    }
};

File: my-folder/0131-palindrome-partitioning/solution.java
class Solution {
    boolean checkPal(String s,int i, int j){
        while(i<=j){
            if(s.charAt(i++)!=s.charAt(j--)) return false;
        }
        return true;
    }
    void getPalindrome(String s,List<List<String>> ans,List<String> temp,int ind){
        if(ind==s.length()){
            ans.add(new ArrayList<>(temp));
            return;
        }
        for(int i=ind;i<s.length();i++){
            if(checkPal(s,ind,i)){
                temp.add(s.substring(ind,i+1));
                getPalindrome(s,ans,temp,i+1);
                temp.remove(temp.size()-1);
            }
        }
    }
    public List<List<String>> partition(String s) {
        List<List<String>> ans= new ArrayList<>();
        List<String> temp = new ArrayList<>();
        getPalindrome(s,ans,temp,0);
        return ans;
        
    }
}
